testnet/stacks-node/src/config.rs

```diff
@@ -506,13 +506,19 @@ lazy_static! {
 }
 
 impl Config {
-    pub fn self_signing(&self) -> Option<SelfSigner> {
+    #[cfg(any(test, feature = "testing"))]
+    pub fn self_signing(&self) -> Option<TestSigners> {
         if !(self.burnchain.mode == "nakamoto-neon" || self.burnchain.mode == "mockamoto") {
             return None;
         }
         self.miner.self_signing_key.clone()
     }
 
+    #[cfg(not(any(test, feature = "testing")))]
+    pub fn self_signing(&self) -> Option<TestSigners> {
+        return None;
+    }
```

Comment:
Should `self_signing` work outside of testing, or should it just be a test only mode for nakamoto?

I could be convinced either way I think. But if we do make self-signing test only, it may be better to just delete mockamoto and the self signing code paths. Then the existing nakamoto tests could just be updated to blindly send signatures to the stackerdb on responses. Thinking it through, I think I'm convincing myself that this would be better. I'll open an issue.

Reply:
That sounds good to me. So we'll leave this as-is and do that in a separate PR?

Reply:
I'm all in favor of deleting self-signing.  I think if we don't delete it, it's going to be the new Helium node -- a vestigial piece of code we're going to be stuck maintaining until the next major architectural revision because someone somewhere might still be using it as part of their devnet setup.